app.rb

```diff
@@ -0,0 +1,185 @@
+    @labels = @data_loader.load_file('labels.json')
+    @music_albums = @data_loader.load_file('music_albums.json')
+    @books = @data_loader.load_file('books.json')
+    @authors = @data_loader.load_file('authors.json')
+    @games = @data_loader.load_file('games.json')
+  end
+
+  def list_books
+    if @books.empty?
+      puts '', 'There are no books', ''
+    else
+      puts '', @books, ''
+    end
+  end
+
+  def list_music_albums
+    if @music_albums.empty?
+      puts("\n- There are no music albums -")
+      return
+    end
+
+    puts 'Music albums:'
+    @music_albums.each_with_index do |music_album, index|
+      puts "#{index + 1}. Title: #{music_album.title} | Publish date: #{music_album.publish_date} |" \
+           "Archived: #{music_album.archived} | On Spotify: #{music_album.on_spotify} | " \
+           "Genre: #{music_album.genre.name}"
+    end
+  end
+
+  def list_games
+    if @games.empty?
+      puts("\n- There are no games -")
+      return
+    end
+
+    puts 'Games:'
+    @games.each_with_index do |game, index|
+      puts "#{index + 1}. Title: #{game.title} | Publish date: #{game.publish_date} | " \
+           "Archived: #{game.archived} | Multiplayer: #{game.multiplayer} | " \
+           "Last played: #{game.last_played_at} | Author: #{game.author.first_name} #{game.author.last_name}"
+    end
+  end
+
+  def list_genres
+    @genres.empty? && puts("\n- There are no genres -")
+
+    puts 'Genres:'
+    @genres.each_with_index do |genre, index|
+      puts "#{index + 1}. #{genre.name}"
+    end
+  end
+
+  def list_labels
+    if @labels.empty?
+      puts '', 'There are no labels', ''
+    else
+      puts('', @labels, '')
+    end
+  end
+
+  def list_authors
+    @authors.empty? && puts("\n- There are no authors -")
+    puts 'Authors:'
+    @authors.each_with_index do |author, index|
+      puts "#{index + 1}. #{author.first_name} #{author.last_name}"
+    end
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+    print 'Publisher: '
+    publisher = gets.chomp
+    print 'Publish date (YYYY/MM/DD): '
+    publish_date = Date.new(*gets.chomp.split('/').map(&:to_i))
+    print 'Cover state [good/bad]: '
+    cover_state = gets.chomp
+    print 'Is it archived? [Y/N]: '
+    archived = gets.chomp.match?(/^[yY]$/)
+    puts 'Please select a label:'
+    list_labels
+    label_id = gets.chomp.to_i
+    relevant_label = @labels.find { |label| label.id == label_id }
+
+    new_book = Book.new(publish_date: publish_date, archived: archived, publisher: publisher, cover_state: cover_state,
+                        title: title)
+    new_book.label = relevant_label
+    @books << new_book
+  end
+
+  def create_music_album
+    # Ask the user for the publish date, on spotify, archived and genre of the music album
+    print "\nPlease enter the title of the album: "
+    title = gets.chomp
+    print 'Please enter the publish date of the album (YYYY/MM/DD): '
```

Comment:
- [ ] [ OPTIONAL] If I enter a wrong date like `1020/50/50` it will crash the whole app it will be better to handle that so when a user enters a wrong date a show a nice msg that says  enter a correct date without crash the app and that way it won't be vulnerable and I think same things for all creation in the app